### ISSUE TITLE
CORS headers not sent in case of errors during authorization code exchange

### ISSUE THREAD

**Issue by syalioune** [keycloak/keycloak#22275]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

authentication

### Describe the bug

When the authorization code exchange for a token fails for some reasons (e.g. : user not allowed to request `offline_access` scope), the `400 Bad Request` response sent to the client does not contain any `CORS` headers even if `Web Origin` is correctly configured on the client.

Tracking down the scenario in the code [here](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java#L1080-L1083), a basic `ErrorResponseException` is thrown instead of `CorsErrorResponseException`.

The consequence is harder root cause analysis because the actual error is masked by the browser because of the `CORS` header absence.

### Version

22.0.1

### Expected behavior

`CORS` headers are correctly sent out to the client as per the client's `Web origins` configuration even for error responses.

### Actual behavior

`CORS` headers are not sent out to the client as per the client's `Web origins` for error responses.
See https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java#L1080-L1083

### How to Reproduce?

1. Create a `test` realm
2. Remove `offline_access` from the realm **default roles**
3. Add a `test` user
4. Create an `angular` client in the `test` realm and enable `Authorization Code flow`
5. Add `*` to `angular` client `Web origin`
6. Clone https://github.com/manfredsteyer/angular-oauth2-oidc/tree/master
7. Adapt `./projects/quickstart-demo/src/app/auth.config.ts`
```ts
import { AuthConfig } from 'angular-oauth2-oidc';

export const authCodeFlowConfig: AuthConfig = {
  issuer: 'http://localhost:8080/realms/test',
  redirectUri: window.location.origin + '/index.html',
  clientId: 'angular',
  responseType: 'code',
  scope: 'openid profile email offline_access',
  showDebugInformation: true,
  timeoutFactor: 0.01,
  checkOrigin: false,
};
```
9. Run `npm install`
10. Run `ng serve --project quickstart-demo -o` and open `http://localhost:4200`
11. Try to connect and authenticate
12. Check with your developer console that `POST http://localhost:8080/realms/test/protocol/openid-connect/token` received a `400` response but is blocked by the browsers because of `CORS`

### Anything else?

The _issue_ is global to `AccessTokenResponseBuilder` subclass.

**Comment by rmartinc** [comment]
@pedroigor @syalioune I cannot reproduce this, even with 22.0.1. The endpoint `token` is not even reached. As soon as I access `http://localhost:4200` the auth endpoint returns this error:

```
http://localhost:4200/index.html?error=invalid_scope&error_description=Invalid+scopes%3A+openid+profile+email+offline_access
```

I cannot even reach the login page.

So please try to re-test this with 24.0.4 and check if you can reproduce with different steps.

**Comment by rmartinc** [comment]
I'm closing this as I could not reproduce the test and it was a long time without any news about this. Please re-open of file a new issue if needed.
